doc/AG0006.md

```diff
@@ -0,0 +1,41 @@
+﻿# Components should have a single public constructor
+
+ID: AG0006
+
+Type: Code Smell
+
+Each component registered with the dependency injection container should have exactly one public constructor. Having multiple public constructors leads to ambiguity in dependency resolution and makes the codebase harder to maintain.
+
+This is a blocker rule - any component with multiple public constructors should be refactored to have only one.
```

Comment:
What does this callout for being "a blocker rule" mean, and is it important? Is it related to severity somehow? Ppl can override it anyway, no?

Reply:
I thik it used to be a problem for DI, but i think it got fixed in like 3.x so im removign this line